ionoscloud/resource_server.go

```diff
@@ -1026,6 +1031,27 @@ func resourceServerUpdate(ctx context.Context, d *schema.ResourceData, meta inte
 	return resourceServerRead(ctx, d, meta)
 }
 
+func deleteAllVolumes(ctx context.Context, d *schema.ResourceData, client *ionoscloud.APIClient, server *ionoscloud.Server) diag.Diagnostics {
+	dcId := d.Get("datacenter_id").(string)
+
+	if entities, ok := server.GetEntitiesOk(); ok {
+		if volumes, ok := entities.GetVolumesOk(); ok {
+			if volumes.Items != nil {
```

Comment:
@gabi-ionos I played with the debugger a little bit, `volumes.Items` is always `nil` here, the volumes are never deleted and hence the dangling volumes from the previous comment.